crates/runtime/src/instance/allocator/memfd.rs

```diff
@@ -0,0 +1,258 @@
+//! memfd mapping logic for use by the pooling allocator.
```

Comment:
as a minor nit I was originally thinking that this and `runtime/src/memfd.rs` would be the same file, in further effort to reduce the number of `#[cfg]` necessary

Reply:
Yes, good point; I've moved everything into one `memfd.rs` in `runtime/src/` (and `memfd_disabled.rs` alongside it).